src/Business Foundation/App/NoSeries/src/Setup/NoSeriesSetupImpl.Codeunit.al

```diff
@@ -15,6 +15,7 @@ codeunit 305 "No. Series - Setup Impl."
         NumberFormatErr: Label 'The number format in %1 must be the same as the number format in %2.', Comment = '%1=No. Series Code,%2=No. Series Code';
         UnIncrementableStringErr: Label 'The value in the %1 field must have a number so that we can assign the next number in the series.', Comment = '%1 = New Field Name';
         NumberLengthErr: Label 'The number %1 cannot be extended to more than 20 characters.', Comment = '%1=No.';
+        CodeFieldChangedErr: Label 'The filter on Series Code was altered by an event subscriber. This is a programming error. Please contact your partner to resolve the issue.\Original Series Code: %1\Modified Filter: %2';
```

Comment:
What about replacing the "Series Code" part with another placeholder, to be filled by line 149 with the current field caption?

Reply:
We could do that. Placeholders make things more complicated for translations. Something the order of words changes depending on the language and if you use placeholders you can't easily fix that. I don't know if that would be an issue here...

Reply:
Just thinking about Spanish or French for example, the translation would put 'original' after series code.  With this change, that is no longer possible.
@NKarolak @PeterDurrer

Reply:
@grobyns Sorry i forget the comments to explain the Placeholders, its know done. I Think the Idea of @NKarolak is good, its better to use the fieldcaption so if the Caption of field is changed you can find the field in the Message otherwise you Searching for the field but it captions is other than in the error.

Reply:
@grobyns: Is it because all translations are automatic only, and there is no productive way to interact?
Usually, I'd explain all placeholders in the comment. The (human) translators then change the order if required: e.g., from `SomeWord %1` to `%1 SomeWord`, depending on the context. We do that for your apps without any problems.

Reply:
@PeterDurrer @NKarolak 
The problem is that a translator (machine or human) needs to know that when translating 'Original %1: %2' to Spanish they need to actually use '%1 Original: %2'

in this case it seems pretty static per language but we have other cases where the order could change based on the value of the placeholders. 
In general we try to avoid:
- composite sentences. Typically you get this when you have the same message multiple times with only small variations. It's better to repeat the whole thing as that makes for much easier translations.
- multiple successive placeholders (like foo %1 %2 %3 bar). Again, it's hard to know whether the order should change.
- when a placeholder can really have only 1 value, then we shouldn't use a placeholder. Here, the %1 would always be 'Series Code'. A comment can be added stating that Series Code is the caption of the "Series Code" field. It's no guarantee but at least the info is there for translators.

Reply:
In any case, these are recommendations. If they were 'laws' we'd have cop rules wouldn't we 🤔 ? I'll leave it to you guys to decide what you think is best.

Reply:
Thanks for the interesting conversation. Finally, I'll show you the link to [Microsoft Best Practices for User Messages](https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/devenv-progress-windows-message-error-and-confirm-methods#best-practices-for-user-messages). It says that it's better to use the field caption if you can. So I'll leave this version. But I can understand both inputs and will think about it in my next code.

Reply:
I think that comment is about using Name v. Caption... and of course you should use Caption.

Reply:
@PeterDurrer I've got feedback from our internal translators: They agree placeholders should be used only if multiple values are possible. My information was outdated - today, automatic translation tools / AI are used first, and they still cannot process the commented placeholders correctly (yet).
So: from the translational point of view, it's indeed better to return to your original draft. Sorry for the previous confusion.